src/shims/env.rs

```diff
@@ -91,19 +176,59 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                     .deallocate(var, None, MiriMemoryKind::Machine.into())?;
             }
             this.update_environ()?;
-            Ok(0)
+            Ok(0) // return zero on success
         } else {
             Ok(-1)
```

Comment:
@christianpoveda where does the -1 here come from? The docs say this should return EINVAL, so shouldn't we look this up in libc instead?

Reply:
from [`man setenv`](https://linux.die.net/man/3/setenv):

```quote
The setenv() function returns zero on success, or -1 on error, with errno set to indicate the cause of the error. 
```

Reply:
Ah so what's missing here is setting errno?

Reply:
Okay @JOE1994 is adding that, nice.